### PR TITLE
fix: [CDS-51458]: Fixed fqnPath issue for ACRArtifactSource

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by pankajsingh-harness** [comment]
retrigger titlecheck

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/artifactSourceUtils.ts [hunk 1]
```diff
@@ -206,23 +206,6 @@ export function getFqnPath(
   }
 }
 
-export function getConnectorRefFqnPath(
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/artifactSourceUtils.ts:209]: Removed this function as this is exactly same as `getFqnPath` defined in line 192.

**Comment by pankajsingh-harness** [comment]
retrigger coverage

**Comment by pankajsingh-harness** [comment]
retrigger coverage

**Comment by pankajsingh-harness** [comment]
retrigger cypress-cv

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx [hunk 2]
```diff
@@ -98,17 +100,32 @@ const Content = (props: ACRRenderContent): JSX.Element => {
...
 
   const serviceId = isNewServiceEnvEntity(path as string) ? serviceIdentifier : undefined
-  const connectorRefFqnPath = getConnectorRefFqnPath(
+  const resolvedArtifactPath = defaultTo(
+    isSidecar
+      ? artifactPath?.split('[')[0].concat(`.${get(initialValues?.artifacts, `${artifactPath}.identifier`)}`)
+      : artifactPath,
+    ''
+  )
+  const subscriptionsFqnPath = getFqnPath(
```
Comment (vikrant-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx:109]: can we make this a generic function and pass the entity name we need to add in fqn ?

Comment (pankajsingh-harness) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx:109]: Sure @vikrant-harness. Let me make this change.

**Review by vikrant-harness — APPROVED** [review]

**Comment by pankajsingh-harness** [comment]
retrigger cypress-pipeline

**Comment by pankajsingh-harness** [comment]
retrigger cypress-pipeline

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

**Pull request merged**